jmclient/jmclient/cryptoengine.py

```diff
@@ -434,6 +434,11 @@ def derive_bip32_pub_export(cls, master_key, path):
         return super(BTC_Watchonly_P2WPKH, cls).derive_bip32_pub_export(
             master_key, BTC_Watchonly_Timelocked_P2WSH.get_watchonly_path(path))
 
+    @classmethod
+    def derive_bip32_priv_export(cls, master_key, path):
+        #return pub key instead of priv key in watchonly wallet
+        return BTC_Watchonly_P2WPKH.derive_bip32_pub_export(master_key, path)
```

Comment:
Maybe add comment here that this is used for wallet id generation, thus we cannot return blank string here and pubkey must be returned instead?

Alternative approach would be to modify `_get_key_ident()` in `jmclient/jmclient/wallet.py` (probably better, as here currently method does not do what its name suggests).